app/lib/core/services/service_locator.dart

```diff
@@ -14,14 +17,21 @@ import 'navigation/nav_service.dart';
 final locator = GetIt.instance;
 
 void setUpLocator() {
+  locator.registerFactory<DatabaseInterface>(
+      () => FirebaseDatabaseInterface(FirebaseFirestore.instance));
+
+  locator.registerLazySingleton(
```

Comment:
- [ ] Shouldn't this be?

```suggestion
  locator.registerLazySingleton<DatabaseService>(
```